addons/mail/models/mail_activity.py

```diff
@@ -203,7 +213,7 @@ def _compute_res_name(self):
     @api.depends('date_deadline')
     def _compute_state(self):
         for record in self.filtered(lambda activity: activity.date_deadline):
-            tz = record.user_id.sudo().tz
+            tz = record.tz
             date_deadline = record.date_deadline
```

Comment:
You will have to align all date computation on stored TZ. Not sure all of this is still correct.